vscode-extension/src/extension.ts

```diff
@@ -615,8 +622,11 @@ async function checkPython() {
  * Checks if pip is installed on the system, and if not, prompts the user to install it.
  */
 async function checkPip() {
+  const pythonPath = await getPythonPath();
   return new Promise((resolve, _reject) => {
-    exec("pip3 --version", (error, stdout, stderr) => {
+    // when calling pip using `python -m`, no need to worry about pip vs pip3. 
+    // You're directly specifying which Python environment's pip to use.
+    exec(pythonPath + " -m pip --version", (error, stdout, stderr) => {
```

Comment:
Pip or pip3?

Reply:
per my comment, installing any python comes with a "pip". invoking with the `-m pip` will need to be pip, but I believe IF pip3 exists, it might work too.